dev/update_requirements.py

```diff
@@ -0,0 +1,118 @@
+"""
+This script updates the `max_major_version` attribute of each package in a YAML dependencies
+specification (e.g. requirements/core-requirements.yaml) to the maximum available version on PyPI.
+"""
+
+import argparse
+import difflib
+import re
+
+import yaml
+import requests
+from packaging.version import Version, InvalidVersion
+
+
+def get_latest_major_version(package_name: str, minium_version: str = None) -> int:
```

Comment:
Nit:

```suggestion
def get_latest_major_version(package_name: str, minimum_version: str = None) -> int:
```

Reply:
I'll remove this argument. https://github.com/mlflow/mlflow/pull/6287#discussion_r929375321